[WIP] Docker team feedback

Pive01 opened the pull request:
- [x] When the user clicks on Start, it starts pulling the localstack image. Why is the notification "warning"? Shouldn't it be a success/info?
> I think a warning message it's correct because it notifies you that you haven't it already and starts pulling it....Success would make sense if the pull was completed I guess (?))
- [ ] I ran out of memory while pulling the image, and I didn't get any messages from the UI. Definitely not a blocker, but if you could improve this, and show the process, it would be great!
- [x] Once I successfully start the extension, what does the Start button do? I don't see any message in the UI after I press it. If it is just for starting up Localstack, maybe you can hide that button based on the "running" status.
> Done in 39e0fcb4c5b517dece3a75f68f6aa68efe605e0b
- [x] There are 2 texts in the header that are not readable with light mode
> Done in #11 
- [x] If I stop Localstack, I lose all my configuration settings, should this happen? If so, this button should be red and have a confirmation
> Done in #9 
- [x] After changing the configuration, I didn't get any feedback from the UI. How does the user know it Is working? Should the user restart in order for that configuration to work?
> Done in #9
- [x] When adding a new configuration and adding some variables, when I press on edit, the variables are not there
> Done in #9
- [x] Inside the edit configuration modal, instead of having a delete option, it's a better practice to have a delete button next to the edit one. This button should be red and have a confirmation dialog
> Done in ff4e15c78bf6c5aa162ff5aea8323645301fa757
- [x] docker extension validate command needs to pass successfully
- [x] The image needs to be built for amd64 and arm64
> Done in 663f0a43097f6231442652754f15c8b44b528786 but not tested
- [x] Extension description details are empty
> Done in cf7df445d4f7146991975c5bb3715ddf23a0b2bc
- [x] Also worth to include in this description / start up screen inside the extension a how to install the awscli-local. Ideally, it could be done from within the extension, or have the instructions directly there, or at least have a link to the docs.

whummer commented on README.md:
```diff
@@ -4,7 +4,7 @@ The LocalStack Extension for Docker Desktop enables developers working with Loca
 LocalStack empowers developers to use over 75+ AWS services locally while helping them simplify their testing and development workflow. LocalStack supports a comprehensive list of APIs, which you can view on our [Feature coverage](https://docs.localstack.cloud/aws/feature-coverage/) page.
 
-![Snap of LocalStack running on Docker Desktop as a Docker Extension](.github/images/localstack-docker-extension.png)
+![Snap of LocalStack running on Docker Desktop as a Docker Extension](.github/images/1-systenStatus.png)
```
nit:
```suggestion
![Snap of LocalStack running on Docker Desktop as a Docker Extension](.github/images/1-systemStatus.png)
```

lukqw reviewed (commented):
Nice set of changes so far! 
Added a couple of questions/remarks.

Pive01 commented on ui/src/components/Header/Menu.tsx:
```diff
@@ -26,7 +26,7 @@ export const LongMenu = () => {
   return (
     <div>
-      <UpdateDialog open={openModal} onClose={() => setOpenModal(false)} />
+      {openModal && <UpdateDialog open={openModal} onClose={() => setOpenModal(false)} />}
```
Not really...the component was still rendered it just was invisible and fired the update every time...not something you really want...

lukqw replied:
Ah makes sense, so the `useEffect` hook within the `UpdateDialog` was fired since the component was rendered invisibly.

Pive01 commented on ui/src/services/hooks/health.ts:
```diff
@@ -23,7 +23,7 @@ export const getDockerStatus = async (): Promise<Health> => {
   const { data, error, isValidating, mutate } = useSWR<Health>(
     SwrCacheKey.HEALTH,
-    () => getDockerStatus(), { revalidateOnMount: true, refreshInterval: 2000 },
+    () => getDockerStatus(), { revalidateOnMount: true },
```
It will update just not constantly......If LocalStack is not running you get an on the console an error every 2 seconds which is quite annoying...There is still the button to refresh manually in case .... maybe I can re-fetch each time the component re-renders?

lukqw replied:
I think in this case we could ignore what's happening on the console, as users are not able to see that anyway (correct me if I am wrong). 

One more thought I had regarding the navbar was to switch these two components:
![image](https://user-images.githubusercontent.com/39307517/208079240-7e26e642-b6fd-4727-b695-2f9a926b4362.png)

This would allow to keep the layout consistent when starting LS with the start button.

lukqw reviewed (approved):
Awesome work! Looking forward to see this in Docker Desktop 🤩 🚀 🎊

whummer reviewed (approved):
🚀🚀

whummer commented:
@Pive01  Should we release the extension under a new version tag? Btw, would be great if we can also start creating a "Change Log" section in the README, with a short summary of the changes of each release.

One more thing - I think we should remove this section from the "Build & Publish LocalStack Docker Extension" Github build, as the image should only get pushed for builds on `main` itself (not for PRs):
```
  pull_request:
    branches: [ main ]
```

Pive01 commented:
> @Pive01 Should we release the extension under a new version tag? Btw, would be great if we can also start creating a "Change Log" section in the README, with a short summary of the changes of each release.
> 
> One more thing - I think we should remove this section from the "Build & Publish LocalStack Docker Extension" Github build, as the image should only get pushed for builds on `main` itself (not for PRs):
> 
> ```
>   pull_request:
>     branches: [ main ]
> ```

I've tagged it with version 0.2.0 ..... Also in the changelog, I've added everything we changed from the last release :+1:

HarshCasper commented:
@Pive01 Can you also update the `README.md` as well ?
